server/config.go

```diff
@@ -257,6 +263,72 @@ func (c *Config) ApplyEnvOverrides() error {
 	return c.applyEnvOverrides("KAPACITOR", "", reflect.ValueOf(c))
 }
 
+func (c *Config) applyEnvOverridesToMap(prefix string, fieldDesc string, mapValue, key, spec reflect.Value) error {
+	// If we have a pointer, dereference it
+	s := spec
+	if spec.Kind() == reflect.Ptr {
+		s = spec.Elem()
+	}
+
+	var value string
+
+	if s.Kind() != reflect.Struct {
+		value = os.Getenv(prefix)
+		// Skip any fields we don't have a value to set
+		if value == "" {
+			return nil
+		}
+
+		if fieldDesc != "" {
+			fieldDesc = " to " + fieldDesc
+		}
+	}
+
+	switch s.Kind() {
+	case reflect.String:
+		mapValue.SetMapIndex(key, reflect.ValueOf(value))
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+
+		var intValue int64
+
+		// Handle toml.Duration
+		if s.Type().Name() == "Duration" {
+			dur, err := time.ParseDuration(value)
+			if err != nil {
+				return fmt.Errorf("failed to apply %v%v using type %v and value '%v'", prefix, fieldDesc, s.Type().String(), value)
+			}
+			intValue = dur.Nanoseconds()
+		} else {
+			var err error
+			intValue, err = strconv.ParseInt(value, 0, s.Type().Bits())
+			if err != nil {
+				return fmt.Errorf("failed to apply %v%v using type %v and value '%v'", prefix, fieldDesc, s.Type().String(), value)
+			}
+		}
+
+		mapValue.SetMapIndex(key, reflect.ValueOf(intValue))
+	case reflect.Bool:
+		boolValue, err := strconv.ParseBool(value)
+		if err != nil {
+			return fmt.Errorf("failed to apply %v%v using type %v and value '%v'", prefix, fieldDesc, s.Type().String(), value)
+
+		}
+		mapValue.SetMapIndex(key, reflect.ValueOf(boolValue))
+	case reflect.Float32, reflect.Float64:
+		floatValue, err := strconv.ParseFloat(value, s.Type().Bits())
+		if err != nil {
+			return fmt.Errorf("failed to apply %v%v using type %v and value '%v'", prefix, fieldDesc, s.Type().String(), value)
+
+		}
+		mapValue.SetMapIndex(key, reflect.ValueOf(floatValue))
+	case reflect.Struct:
+		if err := c.applyEnvOverridesToStruct(prefix, s); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
Nope, I should add that. Is there a place where the logic for array env elements lives?